src/OSPSuite.Presentation/Presenters/Importer/UnitsEditorPresenter.cs

```diff
@@ -44,7 +47,8 @@ public void SelectDimension(string dimension)
       {
          this.DoWithinExceptionHandler(() =>
          {
-            _selectedUnit = (_dimensions.FirstOrDefault(d => d.Name == dimension) ?? _dimensions.First()).DefaultUnit.Name;
+            Dimension = _dimensions.FirstOrDefault(d => d.Name == dimension) ?? _dimensions.First();
```

Comment:
_dimensions.First();

What is _dimensions?

Reply:
_dimensions is the list of authorized dimensions as you mentioned. Now the check is find the FirstOrDefault with this name (so if the name came wrong from the caller it won't crash) and if it was not found (default is returned) then give me the first dimension no matter the name. I am not sure how to optimize in one single line. If the problem is readability as I think, then it might be something like:
`Dimension = _dimensions.FirstOrDefault(d => d.Name == dimension);`
`if (Dimension == null) Dimension = _dimensions.First();`

Reply:
Although I see no big difference in readability between the current code and the one I have just proposed. Maybe some comments?

Reply:
yeah I Was just wondering if _dimensiosn could be empty that's all. If yes, First will throw